keepvariable/keepvariable_core.py

```diff
@@ -587,17 +612,42 @@ def query(
                 value_str = "|".join(values)
                 final_query += tag_query_template.format(field=field, value=value_str)
 
+        # If no query was specified, return all records from the index
+        if not final_query:
+            final_query = "*"
+
         # Query example: "@type:PIPEL @status:{QUEUED|COMPLETED}"
         # Explanation: find all jobs with type field containing 'PIPEL' and status being either 'QUEUED' or 'COMPLETED'
         query_object = Query(final_query)
         if field_to_sort_by:
             query_object.sort_by(field_to_sort_by, asc=asc)
 
         job_docs: list = self.redis.ft(index_name).search(query_object).docs
-        return {job_doc.id: job_doc.json for job_doc in job_docs}
+        return {job_doc.id: self.decode_loaded_value(job_doc.json) for job_doc in job_docs}
 
     def arrlen(self, name: str, path: str) -> Optional[int]:
         return self.redis.json().arrlen(name, path).pop()
 
     def arrappend(self, name: str, path: str, objects: Sequence) -> Optional[int]:
         return self.redis.json().arrappend(name, path, *objects).pop()
+
+    def scan(self, match_string: str, count: int = 50, type_: Optional[str] = None) -> list[str]:
+        """
+        Find saved keys, matching their name with a given glob-style pattern.
+        This command does not block the server, as it is based on a cursor-style iterator.
+
+        https://redis.io/commands/scan/
+
+        :param match_string: string pattern to match keys against, e.g. 'jobs:*'
+        :type match_string: str
+        :param count: how many rows, defaults to 50
+        :type count: int, optional
+        :param type_: filter on specified Redis key type, defaults to None
+        :type type_: Optional[str], optional
+        :return: list of found keys
+        :rtype: list[str]
+        """
+        return list(self.redis.scan_iter(match_string, count, type_))
```

Comment:
Add comment here what scan_iter returns so everyone doesnt need to look it up

Reply:
I think docstring + type hinting already provides all necessary info?